GeneralsMD/Code/GameEngine/Source/GameClient/MessageStream/SelectionXlat.cpp

```diff
@@ -615,6 +615,20 @@ GameMessageDisposition SelectionTranslator::translateGameMessage(const GameMessa
 				break;
 			}
 
+#if !RTS_GENERALS && RETAIL_COMPATIBLE_BUG
+			// TheSuperHackers @info
+			// This fixes the hack in addDrawableToList that restores the exploit that prevents cargo
+			// units from being properly selected when drag-selecting for non-alternate mouse mode where
+			// the same input is used for both selection and issuing enter commands.
+			if (!TheGlobalData->m_useAlternateMouse && isPoint && drawablesThatWillSelect.size() == 2)
```

Comment:
How abot instead of doing all that new stuff, just restore the original EA code with the bugged behavior inside `addDrawableToList` ? Then there is no need to add all this new bugged code.

Reply:
And forgo the fixes to the beacon and stealthed unit selection?